app/Http/Responses/CreatedResponse.php

```diff
@@ -20,11 +20,7 @@ public function __construct(Model $model, $content = '', $status = 201, array $h
         parent::__construct($content, $status, $headers);
 
         if (!$this->headers->has('Location')) {
-            // By default, we will assume our REST APIs will follow the naming convention
-            // identical to our database structure. However, we may override this by specifying
-            // a resourcePath attribute on the model if we want.
-            $location = $model->resourcePath ?? $model->getTable();
-            $this->header('Location', url("$location/{$model->id}"));
+            $this->header('Location', resource_url($model));
```

Comment:
This function seems to be throwing an error for any `POST` request.

Reply:
It should not. But I think I have an idea of the issue at hand: I included new files, but they don't get picked up automatically by the autoloader. Could you try if the following command in `/var/www` (project root) fixes this issue for you?
```
composer dump-autoload
```

Reply:
It might actually fix the Unauthorized messages as well due to a change in the authorization middlewares. Running `composer update` (or just `composer dump-autoload`) should be part of any deployment, I should have thought of that when submitting the PR!

Reply:
Running that command fixed the `POST` methods and I can now create a game, event, block, group, and message. Should this command be added to the README?

It didn't resolve the `DELETE`, `PATCH` or `PUT` methods throwing Unauthorized but I did find out that to get them working you need to put the user's `session_id` as part of the URL and not as part of the body. Once I added the user's session id to the URL, I was able to perform a `DELETE` to delete a blocked player, and delete a message (should that be possible?).

Reply:
The `DELETE` and `PUT` methods do not throw Unauthorized for me. But I noticed that for `GET` requests, either my Insomnia or Laravel transforms the request body to query parameters under the hood (because `GET` apparently doesn't actually support having a request body). I just pushed a commit with a middleware that injects any query parameters into the request body before it gets processed. Now when I use a `GET` request "with body" it is parsed correctly.

However this should not affect `DELETE`, `PATCH` or `PUT`. I tested with the message edit/deletion and they work both using query parameters as well as using the `session_id` as part of the body. Perhaps you made a typo? I can't reproduce your behaviour for these verbs.

Reply:
Also, I have changed the routing of `groups` to fall under the `rooms` resource. So we have routes like this:
```
/api/rooms/{room_id}/groups
/api/rooms/{room_id}/groups/{group_id}
/api/rooms/{room_id}/groups/{group_id}/messages
/api/rooms/{room_id}/groups/{group_id}/messages/{message_id}
```

Note that to get to a specific message, it will require knowledge of three ids. However I believe that was the case already in the old API. I doubt anyone will want to fetch a single specfic message though (usually a whole group).

I have also updated the CreatedResponse class. It can now handle resources that are only accessible as a nested resource (previously the Location header of a message would be something like `/api/messages/{message_id}` while they are only accessible as a sub-resource of a group).